tools/nogo.js

```diff
@@ -1,8 +1,307 @@
+      "/pkg/sentry/fs/dev/random.go": "fix: gvisor.dev/gvisor/pkg/safemem.FromIOReader composite literal uses unkeyed fields",
+      "/pkg/sentry/fs/tmpfs/inode_file.go": "fix: gvisor.dev/gvisor/pkg/sentry/memmap.MappableRange composite literal uses unkeyed fields",
+      "/pkg/sentry/fs/tmpfs/inode_file.go": "fix: gvisor.dev/gvisor/pkg/sentry/memmap.BusError composite literal uses unkeyed fields",
+      "/pkg/sentry/fs/tty/dir.go": "fix: gvisor.dev/gvisor/pkg/sentry/fs.FileOwner composite literal uses unkeyed fields",
+      "/pkg/sentry/kernel/fd_table.go": "fix: gvisor.dev/gvisor/pkg/sentry/fs/lock.LockRange composite literal uses unkeyed fields",
+      "/pkg/sentry/kernel/seccomp.go": "fix: gvisor.dev/gvisor/pkg/bpf.InputBytes composite literal uses unkeyed fields",
+      "/pkg/sentry/kernel/shm/shm.go": "fix: gvisor.dev/gvisor/pkg/sentry/memmap.BusError composite literal uses unkeyed fields",
+      "/pkg/sentry/kernel/shm/shm.go": "fix: gvisor.dev/gvisor/pkg/sentry/memmap.MappableRange composite literal uses unkeyed fields",
+      "/pkg/sentry/kernel/task_context.go": "fix: gvisor.dev/gvisor/pkg/sentry/arch.Stack composite literal uses unkeyed fields",
+      "/pkg/sentry/kernel/task_signals.go": "fix: gvisor.dev/gvisor/pkg/sentry/arch.Stack composite literal uses unkeyed fields",
+      "/pkg/sentry/loader/elf.go": "fix: gvisor.dev/gvisor/pkg/sentry/arch.AuxEntry composite literal uses unkeyed fields",
+      "/pkg/sentry/loader/loader.go": "fix: gvisor.dev/gvisor/pkg/sentry/arch.AuxEntry composite literal uses unkeyed fields",
+      "/pkg/sentry/memmap/memmap.go": "fix: gvisor.dev/gvisor/pkg/sentry/platform.FileRange composite literal uses unkeyed fields",
+      "/pkg/sentry/mm/address_space.go": "fix: gvisor.dev/gvisor/pkg/usermem.AddrRange composite literal uses unkeyed fields",
+      "/pkg/sentry/mm/aio_context.go": "fix: gvisor.dev/gvisor/pkg/sentry/memmap.BusError composite literal uses unkeyed fields",
+      "/pkg/sentry/mm/aio_context.go": "fix: gvisor.dev/gvisor/pkg/sentry/memmap.MappableRange composite literal uses unkeyed fields",
+      "/pkg/sentry/mm/file_refcount_set.go": "fix: gvisor.dev/gvisor/pkg/sentry/platform.FileRange composite literal uses unkeyed fields",
+      "/pkg/sentry/mm/io.go": "fix: gvisor.dev/gvisor/pkg/usermem.AddrRange composite literal uses unkeyed fields",
+      "/pkg/sentry/mm/pma.go": "fix: gvisor.dev/gvisor/pkg/usermem.AddrRange composite literal uses unkeyed fields",
+      "/pkg/sentry/mm/pma.go": "fix: gvisor.dev/gvisor/pkg/safemem.BlockSeqReader composite literal uses unkeyed fields",
+      "/pkg/sentry/mm/pma.go": "fix: gvisor.dev/gvisor/pkg/sentry/memmap.BusError composite literal uses unkeyed fields",
+      "/pkg/sentry/mm/pma.go": "fix: gvisor.dev/gvisor/pkg/sentry/memmap.MappableRange composite literal uses unkeyed fields",
+      "/pkg/sentry/mm/pma.go": "fix: gvisor.dev/gvisor/pkg/usermem.AddrRange composite literal uses unkeyed fields",
```

Comment:
I think it is arguable whether we want to fix these Range types, but we don't have to decide now.

Reply:
Indeed, there's a good argument that these can be exceptions. I wish I could make the exceptions type-based as opposed to file-based, but we can't.

Reply:
I've dropped all this type of analysis.